lib/charms/kafka/v0/client.py

```diff
@@ -0,0 +1,421 @@
+    parser = argparse.ArgumentParser(description="Handler for running a Kafka client")
+    parser.add_argument(
+        "-t",
+        "--topic",
+        help="Kafka topic provided by Kafka Charm",
+        type=str,
+        default="hot-topic",
+    )
+    parser.add_argument(
+        "-u",
+        "--username",
+        help="Kafka username provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        help="Kafka password provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-c",
+        "--consumer-group-prefix",
+        help="Kafka consumer-group-prefix provided by Kafka Charm",
+        type=str,
+    )
+    parser.add_argument(
+        "-s",
+        "--servers",
+        help="comma delimited list of Kafka bootstrap-server strings",
+        type=str,
+    )
+    parser.add_argument(
+        "-x",
+        "--security-protocol",
+        help="security protocol used for authentication",
+        type=str,
+        default="SASL_PLAINTEXT",
+    )
+    parser.add_argument(
+        "-n",
+        "--num-messages",
+        help="number of messages to send from a producer",
+        type=int,
+        default=15,
+    )
+    parser.add_argument(
+        "-r",
+        "--replication-factor",
+        help="replcation.factor for created topics",
+        type=int,
+    )
+    parser.add_argument(
+        "--num-partitions",
+        help="partitions for created topics",
+        type=int,
+    )
+    parser.add_argument("--producer", action="store_true", default=False)
+    parser.add_argument("--consumer", action="store_true", default=False)
+    parser.add_argument("--cafile-path", type=str)
+    parser.add_argument("--certfile-path", type=str)
+    parser.add_argument("--keyfile-path", type=str)
+    parser.add_argument("--mongo-uri", type=str, required=False, default=None)
+    parser.add_argument("--origin", type=str, required=False, default=None)
+
+    args = parser.parse_args()
+    servers = args.servers.split(",")
+    if not args.consumer_group_prefix:
+        args.consumer_group_prefix = f"{args.username}-" if args.username else None
+
+    origin = args.origin if args.origin else get_origin()
+
+    if args.mongo_uri:
+        mongo_client = MongoClient(args.mongo_uri)
+
+        producer_collection = mongo_client[args.topic].get_collection("producer")
+        consumer_collection = mongo_client[args.topic].get_collection("consumer")
+    else:
+        producer_collection = None
+        consumer_collection = None
+
+    client = retrying(
+        lambda: KafkaClient(
+            servers=servers,
+            username=args.username,
+            password=args.password,
+            security_protocol=args.security_protocol,
+            cafile_path=args.cafile_path,
+            certfile_path=args.certfile_path,
+            keyfile_path=args.keyfile_path,
+        ),
+        Exception
+    )
```

Comment:
We need this. When testing sometimes, because of race condition, it was happening that when the process started the broker were shutting down to update the properties (the process starts when the relation is created) and the client would error out when connecting to the kafka cluster. A simple retried logic proved to be effective on handling this.

Reply:
Sorry, in this case I meant 'use tenacity' instead :)